examples/examples_go_test.go

```diff
@@ -179,3 +180,178 @@ func GenerateRandomString(n int) (string, error) {
 
 	return string(ret), nil
 }
+
+func TestLabelsCombinationsGo(t *testing.T) {
+	type testCase struct {
+		name string
+		s1   labelsState
+		s2   labelsState
+	}
+
+	testCases := []testCase{
+		{
+			"can add an empty label",
+			labelsState{
+				DefaultLabels: map[string]string{},
+				Labels:        map[string]string{},
+			},
+			labelsState{
+				DefaultLabels: map[string]string{},
+				Labels:        map[string]string{"x": ""},
+			},
+		},
+		{
+			"convoluted test case found by random-sampling",
+			labelsState{
+				DefaultLabels: map[string]string{"x": "", "y": "s"},
+				Labels:        map[string]string{"x": ""},
+			},
+			labelsState{
+				DefaultLabels: map[string]string{"x": ""},
+				Labels:        map[string]string{"x": "", "y": ""},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			tc.s1.validateTransitionTo(t, tc.s2)
+		})
+	}
+}
+
+func TestRandomLabelsCombinationsGo(t *testing.T) {
+	labelValues := []string{"", "s"} // empty values are conflated with unknowns in TF internals, must test
+
+	labelsValues := []map[string]string{
+		nil,
+		{},
+	}
+
+	for _, label := range labelValues {
+		m := map[string]string{"x": label}
+		labelsValues = append(labelsValues, m)
+	}
+
+	for _, label1 := range labelValues {
+		for _, label2 := range labelValues {
+			m := map[string]string{
+				"x": label1,
+				"y": label2,
+			}
+			labelsValues = append(labelsValues, m)
+		}
+	}
+
+	states := []labelsState{}
+
+	for _, label1 := range labelsValues {
+		for _, label2 := range labelsValues {
+			states = append(states, labelsState{
+				DefaultLabels: label1,
+				Labels:        label2,
+			})
+		}
+	}
+
+	t.Logf("total state space: %v states", len(states))
+	t.Logf("random-sampling 100 state transitions")
+
+	for i := 0; i < 100; i++ {
+		t.Run(fmt.Sprintf("test-%d", i), func(t *testing.T) {
```

Comment:
`""` is interpreted as `fmt.Sprintf("#%d", i)`
```suggestion
		t.Run("", func(t *testing.T) {
```